### PR TITLE
added amqp tracing

### PR DESCRIPTION
**Pull request by raam86**
using opentracing to trace requests throughout

### CODE REVIEW

File: cmd/assaf-http/main.go [hunk 1]
```diff
@@ -134,28 +135,38 @@ func publishToQueue(sessions chan chan session, msgs <-chan message, ctx context
...
 
-	span, _ := opentracing.StartSpanFromContext(ctx, "publish-to-queue")
-	defer span.Finish()
+
 
 	for session := range sessions {
 		sub := <-session
 		select {
 		case msg := <-msgs:
+			sp := opentracing.SpanFromContext(ctx)
```
Comment (raam86) [inline cmd/assaf-http/main.go:144]: create span

File: cmd/assaf-http/main.go [hunk 2]
```diff
@@ -134,28 +135,38 @@ func publishToQueue(sessions chan chan session, msgs <-chan message, ctx context
...
+			defer sp.Finish()
+
+			// Inject the span context into the AMQP header.
+
+
+			amqpMsg := amqp.Publishing{
```
Comment (raam86) [inline cmd/assaf-http/main.go:150]: create amqp message

File: cmd/assaf-http/main.go [hunk 3]
```diff
@@ -134,28 +135,38 @@ func publishToQueue(sessions chan chan session, msgs <-chan message, ctx context
...
+				Headers:         amqp.Table{},
+				ContentType:     "text/plain",
+				ContentEncoding: "",
+				Body:            msg.body,
+				DeliveryMode:    amqp.Transient, // 1=non-persistent, 2=persistent
+				Priority:        0,              // 0-9
+				// a bunch of application/implementation-specific fields
+			}
+			if err := amqptracer.Inject(sp, amqpMsg.Headers); err != nil {
```
Comment (raam86) [inline cmd/assaf-http/main.go:159]: add headers to the amqp request

File: cmd/assaf-http/main.go [hunk 4]
```diff
@@ -134,28 +135,38 @@ func publishToQueue(sessions chan chan session, msgs <-chan message, ctx context
...
-				amqp.Publishing{
-					Headers:         amqp.Table{},
-					ContentType:     "text/plain",
-					ContentEncoding: "",
-					Body:            msg.body,
-					DeliveryMode:    amqp.Transient, // 1=non-persistent, 2=persistent
-					Priority:        0,              // 0-9
-					// a bunch of application/implementation-specific fields
-				},
+				amqpMsg,
```
Comment (raam86) [inline cmd/assaf-http/main.go:169]: send message

File: cmd/assaf-http/main.go [hunk 5]
```diff
@@ -315,6 +325,16 @@ func subscribe(queueName string, sessions chan chan session, consumeRequests cha
 				}
 
 				if consumeRequest.reqId == routingKey {
+					spCtx, _ := amqptracer.Extract(msg.Headers)
```
Comment (raam86) [inline cmd/assaf-http/main.go:328]: after the round trip has finished we need to match the incoming span to the one we sent so we extract it from the message headers.

**Review by kanglais — APPROVED** [review]

**Pull request merged**